modules/renter/download.go

```diff
@@ -191,33 +191,41 @@ func (r *Renter) Download(path, destination string) error {
 
 	// Look up the most recent contract for each host.
 	// NOTE: this assumes that only one contract is made with each host.
-	contracts := make(map[*fileContract]modules.RenterContract)
+	var contractPieces []struct {
+		contract modules.RenterContract
+		pieces   []pieceData
+	}
 	file.mu.RLock()
 	for _, fc := range file.contracts {
```

Comment:
seems a little strange that you chose to keep the `for _, obj := range` construction instead of just switching to the `for i := range` construction, given that you needed to add like 20 lines of code to make it happen.